### PR TITLE
[SPARK-33333][BUILD][2.4] Upgrade Jetty to 9.4.28.v20200408

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade Jetty to 9.4.28.v20200408 like `master` branch (Apache Spark 3.1).

This includes a subset of https://github.com/apache/spark/commit/722369ee5532c94eb7dd0cbda3b7365fc2f52026.

### Why are the changes needed?

To bring the bug fixes.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

File: mllib-local/src/test/scala/org/apache/spark/ml/linalg/MatricesSuite.scala [hunk 1]
```diff
@@ -862,10 +862,10 @@ class MatricesSuite extends SparkMLFunSuite {
     mat.toString(0, 0)
     mat.toString(Int.MinValue, Int.MinValue)
     mat.toString(Int.MaxValue, Int.MaxValue)
-    var lines = mat.toString(6, 50).lines.toArray
+    var lines = mat.toString(6, 50).split('\n')
```
Comment (maropu) [inline mllib-local/src/test/scala/org/apache/spark/ml/linalg/MatricesSuite.scala:865]: Seems like this change is not related to what the PR title says. Just a cleanup?

Comment (dongjoon-hyun) [reply mllib-local/src/test/scala/org/apache/spark/ml/linalg/MatricesSuite.scala:865]: Actually, I added these to fix build failure in the first try failed at https://github.com/apache/spark/pull/30241 . But, it seems that there was something wrong in my local environement at that time. I'll revert this.

**Comment by dongjoon-hyun** [comment]
Thank you, @maropu !

File: mllib-local/src/test/scala/org/apache/spark/ml/linalg/MatricesSuite.scala (hunk 1 above)
Comment (maropu) [reply mllib-local/src/test/scala/org/apache/spark/ml/linalg/MatricesSuite.scala]: Oh, I see.

**Review by maropu — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @maropu . Merged to branch-2.4.
